plugin/evm/upgrade/acp176/acp176.go

```diff
@@ -180,3 +168,11 @@ func scaleExcess(
 	bigExcess.Div(&bigExcess, &bigTarget)
 	return gas.Gas(bigExcess.Uint64())
 }
+
+func mul[T constraints.Unsigned](a, b T) T {
```

Comment:
Perhaps `mulWithUpperBound` or `clippedMultiply`? I dunno, naming is hard, but I think it's worth indicating what it will do on error.

Reply:
I agree that we can rename this + put a comment